tests/compilation_cache_test.py

```diff
@@ -62,6 +66,15 @@ def tearDownModule():
 def increment_event_count(event):
   _counts[event] += 1
 
+def msg_exists_in_logs(msg: str, records: list[logging.LogRecord],
+                       level: int | None = None) -> bool:
+  return any(msg in record.getMessage() for record in records
+             if level is None or level == record.levelno)
+
+def msg_absent_in_logs(msg: str, records: list[logging.LogRecord],
```

Comment:
nit: you could use `self.assertFalse(msg_exists_in_logs)` and then you don't need this function (and might be a bit easier to read)